src/webgpu/util/conversion.ts

```diff
@@ -1694,9 +1719,9 @@ export const kAllUnsignedIntegerScalarsAndVectors = [
 ] as const;
 
 /// All floating-point and integer scalar and vector types
-export const kAllFloatAndIntegerScalarsAndVectors = [
+export const kAllFloatAndConcreteIntegerScalarsAndVectors = [
```

Comment:
Bit confused by this. `AllFloat` includes abstracts? If so, why are integers only concrete?

Reply:
For many builtins AI->AF conversion means that that an AI literal is a valid input type, but the concrete ints (i32 and u32) are not. So often the concrete integers are in the bucket of things that are expected to always fail validation and abstract ints are in the expected to pass validation in some cases bucket with the floats..

Reply:
Ack thanks. Might be worth adding this part of the comment on this `const`. I'll forget in 3 seconds time, and will probably end up asking you again very soon.

Reply:
Added a comment about concrete vs abstract ints